### PR TITLE
[CPU] Introduced shape agnostic eltwise

### PR DESCRIPTION
**Pull request by antonvor**
TODO: leave a more detailed comment

Current limitations: dynamic node, without fused FQ

### CODE REVIEW

**Comment by antonvor** [comment]
@dmitry-gorokhov could you review please?

File: src/plugins/intel_cpu/src/nodes/eltwise.h [hunk 1]
```diff
@@ -35,13 +35,19 @@ struct jit_eltwise_params {
...
 struct jit_eltwise_call_args_ptrs {
     const void *src_ptr[MAX_ELTWISE_INPUTS];
     void *dst_ptr;
     //ptr to array of post op inputs pointers (flat list)
     const void** post_op_data;
+
+    // shape agnostic kernel
+    size_t work_amount;
+    const void *src_offsets[MAX_ELTWISE_INPUTS];
+    const void *dst_offsets;
```
[the remark this answers is not in the document]
Comment (antonvor) [reply src/plugins/intel_cpu/src/nodes/eltwise.h:50]: it's a pointer to multiple offsets, not just one

File: src/plugins/intel_cpu/src/nodes/eltwise.h [hunk 2]
```diff
@@ -66,6 +72,12 @@ struct jit_uni_eltwise_kernel {
     jit_eltwise_params jep_;
 };
 
+enum EltwiseImplType {
```
[the remark this answers is not in the document]
Comment (antonvor) [reply src/plugins/intel_cpu/src/nodes/eltwise.h]: done, thanks

File: src/plugins/intel_cpu/src/nodes/eltwise.cpp [hunk 3]
```diff
@@ -1167,17 +1199,30 @@ struct EltwiseKey {
...
-                      useJit == rhs.useJit;
-
-        for (size_t i = 0; i < inpDims.size() && result; ++i) {
-            result = result && (inpDims[i] == rhs.inpDims[i]);
+                      implType == rhs.implType;
+
+        if (result) {
+            if (implType == optimizedShapeAgnostic) {
+                bool broadcast, rhsBroadcast;
+                for (size_t i = 0; i < inpDims.size() && result; ++i) {
```
[the remark this answers is not in the document]
Comment (antonvor) [reply src/plugins/intel_cpu/src/nodes/eltwise.cpp]: done, thanks

**Comment by antonvor** [comment]
@dmitrygo fyi added motivation and subgraph, could you take a look please?

**Review by dmitry-gorokhov — APPROVED** [review]

**Pull request merged**